File: LibraryAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryAPI.Models
{
public class Book
{
    [Key]
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Author { get; set; }
    public string Description { get; set; }
    public string CoverImage { get; set; }
    public int UserRating { get; set; }
    public string Publisher { get; set; }
    public DateTime PublicationDate { get; set; }
    public string Category { get; set; }
    public string ISBN {  get; set; }
    public int PageCount { get; set; }
    public string[]? Reviews { get; set; }
    public Guid? CheckedOutBy { get; set; }
    public DateTime DueDate { get; set; }

}
}

File: LibraryUI/src/app/shared/models/Book.ts
export class Book {
  id: string;
  title: string;
  coverImage: string;
  author: string;
  rating: number;
  description: string;
  publisher: string;
  publicationDate: any;
  category: string;
  isbn: string;
  pageCount: number;
  reviews: string[]
  checkedOutBy: string;
  dueDate: Date;
  checkOutDisabled: boolean | undefined;

  constructor() {
    this.id = "";
    this.title = "";
    this.coverImage = "";
    this.author = "";
    this.rating = 0;
    this.description = "";
    this.publisher = "";
    this.publicationDate = new Date();
    this.category = "";
    this.isbn = "";
    this.pageCount = 0;
    this.reviews = []
    this.checkedOutBy = "";
    this.dueDate = new Date();
    this.checkOutDisabled = undefined;
  }

}
